aesthetic/avg_autotags.py

```diff
@@ -23,51 +23,80 @@
     # e.g. 'Upper Lawrenceville': Counter({'indoor': 28.342, 'text': 20.451, ...})
     nghds_autotags = collections.defaultdict(collections.Counter)
 
+    # this is the final output dict with neighborhood: 'num_photos': x , 'autotags_90plus_minusbaseline': [{'autotag': x, 'score': x, 'examples': x}, ]
+    nghd_autotags_urls = dict()
     overall_autotags_90plus = collections.Counter()
     for row in csv.reader(open(args.yfcc_autotags_file)):
-        photo_id, nsid, lat, lon = row[0:4]
+        photo_id, nsid, lat, lon, url = row[0:5]
         nghd = pgh_pointmap[lat, lon]
-
-        autotags_90plus = ast.literal_eval(row[4])
-        autotags = row[5].split(',')
+        # autotags_90plus is the autotags for each pic with >=0.9 score
+        autotags_90plus = ast.literal_eval(row[5])
+        autotags = row[6].split(',')
         autotags_ctr = collections.Counter()
+        if nghd not in nghd_autotags_urls: nghd_autotags_urls[nghd] = dict()
         for tag in autotags:
             if tag == '':
                 continue
             else:
                 tagname, value = tag.split(':')
-                nghds_autotags[nghd][tagname] += round(float(value), 3)
+                nghds_autotags[nghd][tagname] += round(float(value), 3) 
+                if tagname in nghd_autotags_urls[nghd]: 
+                    nghd_autotags_urls[nghd][tagname] += [url]
+                else: 
+                    nghd_autotags_urls[nghd][tagname] = [url]
+        #'NUM_PHOTOS' is number of photos of that neighborhood            
         nghds_autotags[nghd]['NUM_PHOTOS'] += 1
+        #put in 'NUM_PHOTOS' to nghds_autotags90plus
         nghds_autotags90plus[nghd].update(autotags_90plus + ['NUM_PHOTOS'])
+        #put in 'NUM_PHOTOS', autotags_90plus to overall_autotags_90plus
         overall_autotags_90plus.update(autotags_90plus + ['NUM_PHOTOS'])
- 
-    
+        # this looks like Counter({'NUM_PHOTOS': 15, 'outdoor': 5, 'building':
+        # 2, 'gable': 2, 'nature': 2, 'indoor': 2, 'plant': 2, 'architecture':
+        # 2, 'pink': 1, 'vehicle': 1, 'house': 1, 'pediment': 1, 'frieze': 1,
+        # 'flower': 1, 'green': 1, 'suit': 1, 'banana tree': 1, 'car': 1})
+
     # Compute the overall rates of each tag, so we can subtract them off.
     tempdict = collections.Counter()
     for tag, value in overall_autotags_90plus.items():
         tempdict[tag] = (value*1.0/overall_autotags_90plus['NUM_PHOTOS'])
     del tempdict['NUM_PHOTOS']
     overall_autotags_90plus = tempdict
-
-    output = {}
+    sorted_output = dict()
+    output = dict()
```

Comment:
again, just delete my `output` variable, and replace it with your `sorted_output`. Don't worry, I won't mind :)